clients/drcachesim/tests/trace_interval_analysis_unit_tests.cpp

```diff
@@ -85,15 +92,42 @@ class test_stream_t : public scheduler_t::stream_t {
     scheduler_t::input_ordinal_t
     get_input_stream_ordinal() override
     {
-        return 0;
+        assert(at_ > 0);
+        // All TIDs form a separate input stream.
+        return refs_[at_ - 1].instr.tid;
```

Comment:
It feels weird to always be referencing -1 especially when the name "at_" sounds like it should point to the current record: better to have at_ be signed and start -1 and point to the current record instead of one past it?

Reply:
Done.